diff --git a/metatag_open_graph/src/Plugin/metatag/Tag/OgLocale.php b/metatag_open_graph/src/Plugin/metatag/Tag/OgLocale.php
index 9b3905e..9a924fb 100644
--- a/metatag_open_graph/src/Plugin/metatag/Tag/OgLocale.php
+++ b/metatag_open_graph/src/Plugin/metatag/Tag/OgLocale.php
@@ -10,7 +10,7 @@ use Drupal\metatag\Plugin\metatag\Tag\MetaPropertyBase;
  * @MetatagTag(
  *   id = "og_locale",
  *   label = @Translation("Locale"),
- *   description = @Translation("The locale these tags are marked up in, must be in the format language_TERRITORY. Default is 'en_US'."),
+ *   description = @Translation("This field is used in SDP to set the page’s language, display font and text direction. See our <a href='https://digital-vic.atlassian.net/servicedesk/customer/portal/14/article/2272657746'>full list of language codes</a>."),
  *   name = "og:locale",
  *   group = "open_graph",
  *   weight = 26,
