diff --git a/src/JiraRestWrapperService.php.old b/src/JiraRestWrapperService.php
index 390ac8d7..d000cbc0 100644
--- a/src/JiraRestWrapperService.php.old
+++ b/src/JiraRestWrapperService.php
@@ -7,6 +7,7 @@
 use Drupal\key\KeyRepositoryInterface;
 use JiraRestApi\Configuration\ArrayConfiguration;
 use JiraRestApi\Issue\IssueService;
+use JiraRestApi\User\UserService;
 use JiraRestApi\JiraException;
 
 /**
@@ -102,4 +103,35 @@ public function getIssueService($endpoint_id = NULL) {
     }
     return $issueService;
   }
+
+  /**
+   * Get the User service api.
+   *
+   * @return \JiraRestApi\User\UserService
+   *   User Service API.
+   * @throws \JiraRestApi\JiraException
+   * @throws \JsonMapper_Exception
+   * @throws \Exception
+   */
+  public function getUserService($endpoint_id = NULL) {
+    // Attempt to get a specific endpoint
+    if (!empty($endpoint_id) ) {
+      $endpoint = $this->endpointRepository->getEndpoint($endpoint_id);
+    }
+    if (!isset($endpoint)) {
+      $endpoint = $this->endpointRepository->getDefaultEndpoint();
+    }
+
+    if (empty($endpoint)) {
+      throw new JiraException($this->t('No JIRA Endpoints could be found.'));
+    }
+
+    // Initialize the JIRA User Service
+    try {
+      $userService = new UserService($this->getArrayConfiguration($endpoint));
+    } catch (JiraException $e) {
+      $this->loggerRestJira->error($e->getMessage());
+    }
+    return $userService;
+  }
 }
