diff --git a/src/Plugin/diff/Field/AddressFieldBuilder.php b/src/Plugin/diff/Field/AddressFieldBuilder.php
--- a/src/Plugin/diff/Field/AddressFieldBuilder.php
+++ b/src/Plugin/diff/Field/AddressFieldBuilder.php
@@ -21,7 +21,7 @@
   /**
    * {@inheritdoc}
    */
-  public function build(FieldItemListInterface $field_items) {
+  public function build(FieldItemListInterface $field_items): array {
     $result = [];
 
     foreach ($field_items as $field_key => $field_item) {
@@ -30,7 +30,7 @@
           'label' => 'hidden',
           'type' => 'address_plain',
         ]);
-        $rendered_value = $this->renderer->renderPlain($value);
+        $rendered_value = $this->renderer->renderInIsolation($value);
         $result[$field_key][] = $rendered_value;
       }
     }
