diff --git a/src/Plugin/Validation/Constraint/PasswordPolicyConstraintValidator.php b/src/Plugin/Validation/Constraint/PasswordPolicyConstraintValidator.php
--- a/src/Plugin/Validation/Constraint/PasswordPolicyConstraintValidator.php
+++ b/src/Plugin/Validation/Constraint/PasswordPolicyConstraintValidator.php
@@ -13,13 +13,6 @@ use Symfony\Component\DependencyInjection\ContainerInterface;
  */
 class PasswordPolicyConstraintValidator extends ConstraintValidator implements ContainerInjectionInterface {

-  /**
-   * Validator 2.5 and upwards compatible execution context.
-   *
-   * @var \Symfony\Component\Validator\Context\ExecutionContextInterface
-   */
-  protected $context;
-
   /**
    * {@inheritdoc}
    */
